File: 백준/Silver/15803. PLAYERJINAH’S BOTTLEGROUNDS/PLAYERJINAH’S BOTTLEGROUNDS.py
import sys
input = sys.stdin.readline

p = [tuple(map(int, input().split())) for _ in range(3)]

if p[0][0] == p[1][0] == p[2][0]:
    print('WHERE IS MY CHICKEN?')
    quit()
if p[0][1] == p[1][1] == p[2][1]:
    print('WHERE IS MY CHICKEN?')
    quit()
    
m11 = abs(p[1][0]-p[0][0])
m12 = abs(p[1][1]-p[0][1])
m21 = abs(p[2][0]-p[1][0])
m22 = abs(p[2][1]-p[1][1])
m31 = abs(p[2][0]-p[0][0])
m32 = abs(p[2][1]-p[0][1])

if 0 in [m11, m12, m21, m22, m31, m32]:
    print('WINNER WINNER CHICKEN DINNER!')
    quit()

a1 = m12 / m11
a2 = m22 / m21
a3 = m32 / m31

if a1 == a2 == a3:
    print('WHERE IS MY CHICKEN?')
else:
    print('WINNER WINNER CHICKEN DINNER!')

File: 백준/Silver/15803. PLAYERJINAH’S BOTTLEGROUNDS/PLAYERJINAH’S BOTTLEGROUNDS.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "./example.txt")
  .toString()
  .trim()
  .split("\n")
  .map((v) => v.trim().split(" ").map(Number));

let [...p] = input;

function calc(a, b) {
  let r1 = Math.abs(a[0] - b[0]);
  let r2 = Math.abs(a[1] - b[1]);

  if (r1 === 0 || r2 === 0) {
    return 0;
  } else {
    return r2 / r1;
  }
}

const res1 = calc(p[0], p[1]);
const res2 = calc(p[1], p[2]);
const res3 = calc(p[0], p[2]);

if (res1 === res2 && res2 === res3 && res1 === res3) {
  console.log("WHERE IS MY CHICKEN?");
} else {
  console.log("WINNER WINNER CHICKEN DINNER!");
}
